### PR TITLE
docs(deployment): sync design to online docs

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Description

Fixes # (issue)

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: docs/en/latest/architecture-design/deployment-role.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+## Concept
```
Comment (SylviaBABY) [inline docs/en/latest/architecture-design/deployment-role.md:24]:
```suggestion
##  Background
```

File: docs/en/latest/architecture-design/deployment-role.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+Previously, the DP (data plane) and the CP (control plane) are not separate explicitly.
+Although we clearly distinguish the different responsibilities of DP and CP in the documentation, not everyone has correctly deployed APISIX in the production environment.
```
Comment (SylviaBABY) [inline docs/en/latest/architecture-design/deployment-role.md:27]:
```suggestion
Previously, the DP (Data Plane) and the CP (Control Plane) are not separate explicitly.

Although we clearly distinguish the different responsibilities of DP and CP in the documentation, not everyone has correctly deployed APISIX in the production environment.
```

File: docs/en/latest/architecture-design/deployment-role.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+Therefore, we introduce new concepts called deployment modes/roles, to help users deploy APISIX easily and safely.
+APISIX under different deployment modes will act differently.
```
Comment (SylviaBABY) [inline docs/en/latest/architecture-design/deployment-role.md:30]:
```suggestion
Therefore, we introduce new concepts called deployment modes/roles, to help users deploy APISIX easily and safely.

APISIX under different deployment modes will act differently.
```

File: docs/en/latest/architecture-design/deployment-role.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+## Deployment Modes
+
+### Traditional
+
+![traditional](../../../assets/images/deployment-traditional.svg)
+
+In the traditional deployment mode, one instance can be both DP & CP.
+There will be a `conf server` listens on Unix socket and acts as a proxy between APISIX and etcd.
+Both the DP part and CP part of the instance will connect to the `conf server` via HTTP protocol.
```
Comment (SylviaBABY) [inline docs/en/latest/architecture-design/deployment-role.md:48]:
```suggestion
In the traditional deployment mode, one instance can be both DP & CP.

There will be a `conf server` listens on UNIX socket and acts as a proxy between APISIX and etcd.

Both the DP part and CP part of the instance will connect to the `conf server` via HTTP protocol.
```

File: docs/en/latest/architecture-design/deployment-role.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+## Concept
+
+Previously, the DP (data plane) and the CP (control plane) are not separate explicitly.
```
Comment (SylviaBABY) [inline docs/en/latest/architecture-design/deployment-role.md:26]:
```suggestion
Previously, the DP (Data Plane) and the CP (Control Plane) are not separate explicitly.

```

File: docs/en/latest/architecture-design/deployment-role.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+| standalone       | data_plane                 | Only DP, load the all configurations from local yaml file                                |
+
+## Deployment Modes
+
+### Traditional
+
+![traditional](../../../assets/images/deployment-traditional.svg)
+
+In the traditional deployment mode, one instance can be both DP & CP.
+There will be a `conf server` listens on Unix socket and acts as a proxy between APISIX and etcd.
```
Comment (SylviaBABY) [inline docs/en/latest/architecture-design/deployment-role.md:47]:
```suggestion

There will be a `conf server` listens on UNIX socket and acts as a proxy between APISIX and etcd.

```

File: docs/en/latest/architecture-design/deployment-role.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+       timeout: 30
+```
+
+### Decoupled
+
+![decoupled](../../../assets/images/deployment-cp_and_dp.svg)
+
+The instance deployed as data_plane will:
+
+1. Fetch configurations from control plane, the default port is 9280
```
Comment (SylviaBABY) [inline docs/en/latest/architecture-design/deployment-role.md:70]:
```suggestion
1. Fetch configurations from the CP, the default port is 9280
```

Comment (SylviaBABY) [reply docs/en/latest/architecture-design/deployment-role.md:70]: cause all the usages in the previous contents are DP and CP, recommended to change to CP here.

File: docs/en/latest/architecture-design/deployment-role.md [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+|------------------|----------------------------|------------------------------------------------------------------------------------------|
+| traditional      | traditional                | DP + CP are deployed together by default. People need to disable `enable_admin` manually |
+| decoupled        | data_plane / control_plane | DP and CP are deployed independently.                                                    |
+| standalone       | data_plane                 | Only DP, load the all configurations from local yaml file                                |
+
+## Deployment Modes
+
+### Traditional
+
+![traditional](../../../assets/images/deployment-traditional.svg)
```
Comment (hf400159) [inline docs/en/latest/architecture-design/deployment-role.md:46]: Please do not use a transparent background as the base color of the image.The picture will not be clear when enlarged.
image

Comment (spacewander) [reply docs/en/latest/architecture-design/deployment-role.md:46]: I don't have the non-transparent version of the image. If there is not a simple way to add background, what about removing them?

Comment (SylviaBABY) [reply docs/en/latest/architecture-design/deployment-role.md:46]: I have edited those with white background, you can use that.

**Review by tzssangglass — APPROVED** [review]

**Review by SylviaBABY — APPROVED** [review]

**Pull request merged**